src/screens/BeverageDetailsScreen.js

```diff
@@ -1,41 +1,50 @@
 // @flow
 
-import type { Beverage, EntityID, LoadObject } from 'brewskey.js-api';
+import type { Beverage, EntityID } from 'brewskey.js-api';
 import type { Navigation } from '../types';
 
 import * as React from 'react';
 import InjectedComponent from '../common/InjectedComponent';
-import DAOApi from 'brewskey.js-api';
+import { BeverageStore } from '../stores/DAOStores';
 import { observer } from 'mobx-react';
 import { Text } from 'react-native';
 import Container from '../common/Container';
 import Header from '../common/Header';
-import loadDAOEntity from '../common/loadDAOEntity';
-import withLoadingActivity from '../common/withLoadingActivity';
+import LoaderComponent from '../common/LoaderComponent';
+import LoadingIndicator from '../common/LoadingIndicator';
 import flatNavigationParamsAndScreenProps from '../common/flatNavigationParamsAndScreenProps';
 
 type InjectedProps = {|
-  entityLoader: LoadObject<Beverage>,
   id: EntityID,
   navigation: Navigation,
 |};
 
 @flatNavigationParamsAndScreenProps
 @observer
-@loadDAOEntity(DAOApi.BeverageDAO)
-@withLoadingActivity()
 class BeverageDetailsScreen extends InjectedComponent<InjectedProps> {
-  render() {
-    const { entityLoader } = this.injectedProps;
-    const { description, name } = entityLoader.getValueEnforcing();
+  _renderLoading = (): React.Node => (
```

Comment:
I'll merge this in but I think it would be a lot cleaner if you used pure functional components instead of doing these sub-render functions.